thor/src/lib.rs

```diff
@@ -563,13 +563,18 @@ impl Channel {
             .transpose()
     }
 
-    /// Recycle a channel.
+    /// Splice a channel.
     ///
     /// Create a new funding transaction using a previous funding transaction as
-    /// input This is useless in this state :]
-    pub async fn recycle<T, W>(self, transport: &mut T, wallet: &W) -> anyhow::Result<Self>
+    /// input. Also inject own funds to channel by passing a splice-in amount.
+    pub async fn splice<T, W>(
+        self,
+        transport: &mut T,
+        wallet: &W,
+        splice_in: Option<Amount>,
```

Comment:
I thought about it. It's not that straightforward because we still need to need:

```
#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
#[derive(Clone, Debug)]
pub(crate) enum Splice {
    In {
        #[cfg_attr(
            feature = "serde",
            serde(with = "bitcoin::util::amount::serde::as_sat")
        )]
        amount: Amount,
        #[cfg_attr(
            feature = "serde",
            serde(with = "crate::serde::partially_signed_transaction")
        )]
        input_psbt: PartiallySignedTransaction,
    },
    Out(TxOut),
    None,
}
```

The `psbt`s need to be exchanged as part of the protocol.
Not sure how realistic it is to expect one party to pass the `TxOut` of another party in.

The only valuable information would be the amounts but they're pretty much useless by themselves.

Do you think there is gain in passing amounts?

Reply:
> Do you think there is gain in passing amounts?

I guess that way you can verify that the output in the PSBT they provide pays the right amount. In fact, each party should be able to pre-compute the following before exchanging PSBTs:

1. Amount in the shared fund output: `current_total_balance + splice_in_self + splice_in_other`.
2. Amount in the commit transaction output: `current_total_balance + splice_in_self + splice_in_other - fees`.
3. Amount in `TX_s_output_self`: `current_balance_self + splice_in_self - fees`.
4. Amount in `TX_s_output_other`: `current_balance_other + splice_in_other - fees`.

The PSBTs can then be exchanged to show what inputs and change outputs each party needs to add to compute the same transaction.